utils_cv/action_recognition/dataset.py

```diff
@@ -0,0 +1,473 @@
+        while len(clip) < self.sample_length:
+            clip.append(clip[-1].copy())
+
+        return clip
+
+    def __getitem__(self, idx: int) -> Tuple[torch.tensor, int]:
+        """
+        Return:
+            clips (torch.tensor), label (int)
+        """
+        record = self.video_records[idx]
+        video_reader = decord.VideoReader(
+            "{}.{}".format(
+                os.path.join(self.root, record.path), self.video_ext
+            ),
+            # TODO try to add `ctx=decord.ndarray.gpu(0) or .cuda(0)`
+        )
+        record._num_frames = len(video_reader)
+
+        offsets = self._sample_indices(record)
+        clips = np.array([self._get_frames(video_reader, o) for o in offsets])
+
+        if self.num_segments == 1:
+            # [T, H, W, C] -> [C, T, H, W]
+            return self.transforms(torch.from_numpy(clips[0])), record.label
+        else:
+            # [S, T, H, W, C] -> [S, C, T, H, W]
+            return (
+                torch.stack(
+                    [self.transforms(torch.from_numpy(c)) for c in clips]
+                ),
+                record.label,
+            )
+
+    def _show_batch(
+        self,
+        batch: List[torch.tensor],
+        sample_length: int,
+        mean: Tuple[int, int, int] = DEFAULT_MEAN,
+        std: Tuple[int, int, int] = DEFAULT_STD,
+    ) -> None:
+        """
+        Display a batch of images.
+
+        Args:
+            batch: List of sample (clip) tensors
+            sample_length: Number of frames to show for each sample
+            mean: Normalization mean
+            std: Normalization std-dev
+        """
+        batch_size = len(batch)
+        plt.tight_layout()
+        fig, axs = plt.subplots(
+            batch_size,
+            sample_length,
+            figsize=(4 * sample_length, 3 * batch_size),
+        )
+
+        for i, ax in enumerate(axs):
+            if batch_size == 1:
+                clip = batch[0]
+            else:
+                clip = batch[i]
+            clip = Rearrange("c t h w -> t c h w")(clip)
+            if not isinstance(ax, np.ndarray):
+                ax = [ax]
+            for j, a in enumerate(ax):
+                a.axis("off")
+                a.imshow(
+                    np.moveaxis(denormalize(clip[j], mean, std).numpy(), 0, -1)
+                )
+            pass
+
+    def show_batch(
+        self, train_or_test: str = "train", num_samples: int = 1
+    ) -> None:
+        """Plot first few samples in the datasets"""
+        if train_or_test == "train":
+            batch = [self.train_ds.dataset[i][0] for i in range(num_samples)]
+        elif train_or_test == "valid":
+            batch = [self.test_ds.dataset[i][0] for i in range(num_samples)]
+        else:
+            raise ValueError("Unknown data type {}".format(which_data))
+
+        self._show_batch(
+            batch, self.sample_length, mean=DEFAULT_MEAN, std=DEFAULT_STD,
```

Comment:
[minor] dont think this is necessary: mean=DEFAULT_MEAN, std=DEFAULT_STD

Reply:
I don't think we should expose these two parameters in the function: mean and std since these are "fixed" for training, visualization, etc.

Reply:
it lkooks like i didn't upload my changes to this file... let me do that again